lib/dotcom/trip_plan/alerts.ex

```diff
@@ -35,33 +47,66 @@ defmodule Dotcom.TripPlan.Alerts do
     |> Enum.uniq()
   end
 
-  defp leg_entities(%Leg{mode: mode} = leg) do
+  def leg_entities(%Leg{mode: mode} = leg) do
     for entity <- mode_entities(mode),
         stop_id <- Leg.stop_ids(leg) do
       %{entity | stop: stop_id}
     end
   end
 
-  defp mode_entities(%TransitDetail{route: route, trip: %{id: trip_id}}) do
-    trip =
-      if is_nil(route.external_agency_name) do
-        Schedules.Repo.trip(trip_id)
-      end
+  defp leg_entities_from(%Leg{mode: mode} = leg) do
+    for entity <- mode_entities(mode),
+        stop_id <- Leg.stop_ids_from(leg) do
+      %{entity | stop: stop_id}
+    end
+  end
 
-    route_type =
-      if route do
-        route.type
-      end
+  defp leg_entities_to(%Leg{mode: mode} = leg) do
+    for entity <- mode_entities(mode),
+        stop_id <- Leg.stop_ids_to(leg) do
+      %{entity | stop: stop_id}
+    end
+  end
```

Comment:
Maybe there's a way to reuse the logic from `leg_entities/1` instead of duplicating it?

Reply:
What do you think of [this](https://github.com/mbta/dotcom/pull/2269/commits/e293a16264a8e9a7fd306a5585bbbd796a71c3ad)?

(I think this also addresses https://github.com/mbta/dotcom/pull/2269#discussion_r1889205959)